dashboard/src/modules/containers/MainLayout/index.jsx

```diff
@@ -1,20 +1,48 @@
 import React from "react";
+import { Outlet, useLocation } from "react-router-dom";
 import ToastComponent from "modules/components/ToastNotificationComponent";
 import { useSelector } from "react-redux";
-import { Page, Spinner } from "@patternfly/react-core";
-import NavbarDrawer from "modules/components/NavbarDrawerComponent";
+import HeaderComponent from "modules/components/HeaderComponent";
+import {
+  Spinner,
+  Page,
+  PageSection,
+  PageSectionVariants,
+} from "@patternfly/react-core";
 import Sidebar from "modules/components/SidebarComponent";
 import TableWithFavorite from "modules/components/TableComponent";
+import "./index.less";
+
+const LoginView = () => {
+  const { pathname } = useLocation();
+  return (
+    <>
+      {pathname === "/" ? (
```

Comment:
Can you help me understand what `useLocation()` returns here?  It seems this might be a bit fragile, and we should have a "is logged in" function that should be used instead of the return value from `useLocation()`.

Reply:
The `useLocation ` hook is a function that returns the location object that contains information about the current URL. Whenever the URL changes, a new location object will be returned.

Here, if the user is not logged in  and the URL is "/" it is a browsing page for other URLs login prompt message is shown.

At line:44, `is logged in` is checked and `LoginView` is rendered

Reply:
See https://github.com/distributed-system-analysis/pbench/pull/2817#pullrequestreview-1007673357.